lms/static/js/spec/courseware/course_home_events_spec.js

```diff
@@ -21,19 +21,5 @@ define(['jquery', 'logger', 'js/courseware/course_home_events'], function($, Log
             $('.date-summary-link').click();
             expect(Logger.log).toHaveBeenCalledWith('edx.course.enrollment.upgrade.clicked', {location: 'sidebar'});
         });
-
-        it('sends an event when "Upgrade Now" is clicked from the upsell notification', function() {
-            $('.upgrade-banner-button').click();
-            expect(Logger.log).toHaveBeenCalledWith(
-                'edx.course.enrollment.upgrade.clicked', {location: 'notification'}
```

Comment:
@dianakhuang @marcotuts: I may have another PR on my hands related to this code.  Clearly this location wasn't anything real.  But, I discovered two issues:
1. The hero isn't represented as a location.  You can see the code for a [unique event for it](https://github.com/edx/edx-platform/blob/master/openedx/features/course_experience/static/course_experience/js/CourseHome.js#L146) here.
2. The way the hero event is coded, it will fire for any `btn-upgrade` event, so it will now include all the btn-upgrade buttons.

Some potential solutions:
1. Have the [promotion shared code](https://github.com/edx/edx-platform/blob/master/openedx/features/course_experience/static/course_experience/js/CourseHome.js#L62) also take care of 'edx.course.enrollment.upgrade.clicked', so we always get both together.
2. Separately remove or fix the hero events of they are still needed: 'edx.course.upgrade.hero.clicked' and 'edx.course.upgrade.hero.displayed'

Reply:
@grantgoodmanedX: I'm not sure what the final impact will be of my comment above until the dust settles.  Then I can let you know if any doc updates are needed.
FYI: @catong

Reply:
I would do both 1 and 2 in the solutions, does that make sense? And specifically, I dont know if we need the hero events if they're represented as a general upgrade event with a location

Reply:
@marcotuts: 
1. It turns out that the promotion code is specific to course home, but 'edx.course.enrollment.upgrade.clicked' is used in other places (like learner dashboard).  Also, since the 'edx.course.enrollment.upgrade.clicked' event is documented, there is an advantage to someone explicitly sending it.  It's not perfect, but that's what I went for in adding one for the 'hero'.
2. I switched these events to BI events.  I did not hear that they could be removed, but this at least gets the new naming convention.